src/main/java/org/apache/sysds/runtime/transform/encode/Encoder.java

```diff
@@ -145,7 +145,7 @@ public Encoder subRangeEncoder(int colStart, int colEnd) {
 	 */
 	protected void mergeColumnInfo(Encoder other, int col) {
 		// update number of columns
-		_clen = Math.max(_colList.length, col - 1 + other.getNumCols());
+		_clen = Math.max(_clen, col - 1 + other._clen);
```

Comment:
This was my mistake in the last PR `_colList.length` is something different to `_clen` and `.getNumCols()` will be wrong once we add dummycoding (it gives us the cols after encoding).